core/iwasm/libraries/wasi-nn/test/testapp.c

```diff
@@ -0,0 +1,260 @@
+}
+
+void
+my_compute(graph_execution_context context)
+{
+    printf("running model \n");
+    compute(context);
+}
+
+void
+my_allocate(graph graph)
+{
+    printf("Allocating tensors \n");
+    init_execution_context(graph);
+}
+
+void
+my_output(graph_execution_context context, uint32_t index, float *out_tensor,
+          int out_size)
+{
+    printf("Retrieving output \n");
+    get_output(context, index, out_tensor, out_size);
+}
+
+float *
+my_inference(float *input, uint32_t *input_size, int *output_size,
+             char *model_name)
+{
+    graph graph;
+    printf("end to end  \n");
+
+    my_load(model_name);
+
+    my_allocate(graph);
+
+    my_input(input, input_size);
+
+    graph_execution_context context;
+    my_compute(context);
+
+    uint32_t index;
+
+    float *out_tensor = (float *)malloc(sizeof(float) * 5);
+
+    my_output(context, index, out_tensor, 5 * sizeof(float));
+
+    printf("finished\n");
+
+    return out_tensor;
+}
+
+void
+test_sum()
+{
+    int output_size = 0;
+
+    uint32_t *dim = malloc(4 * sizeof(uint32_t));
+
+    dim[0] = 1;
+    dim[1] = 5;
+    dim[2] = 5;
+    dim[3] = 1;
+
+    float *input_tensor = malloc(INPUT_SIZE * sizeof(float));
+
+    for (int i = 0; i < INPUT_SIZE; i++) {
+
+        input_tensor[i] = 1;
+    }
+
+    float *output =
+        my_inference(input_tensor, dim, &output_size, "models/sum.tflite");
+
+    assert(abs(output[0] - 25.0) < EPSILON);
+}
+
+void
+test_max()
+{
+    int output_size = 0;
+    uint32_t *dim = malloc(4 * sizeof(uint32_t));
+
+    dim[0] = 1;
+    dim[1] = 5;
+    dim[2] = 5;
+    dim[3] = 1;
+
+    float *input_tensor = malloc(INPUT_SIZE * sizeof(float));
+
+    for (int i = 0; i < INPUT_SIZE; i++) {
+
+        input_tensor[i] = i;
+    }
+
+    float *output =
+        my_inference(input_tensor, dim, &output_size, "models/max.tflite");
+
+    printf("max is: %f \n ", output[0]);
+
+    assert(abs(output[0] - 24.0) < EPSILON);
+}
+
+void
+test_average()
+{
+    int output_size = 0;
+    uint32_t *dim = malloc(4 * sizeof(uint32_t));
+
+    dim[0] = 1;
+    dim[1] = 5;
+    dim[2] = 5;
+    dim[3] = 1;
+
+    float *input_tensor = malloc(INPUT_SIZE * sizeof(float));
+
+    for (int i = 0; i < INPUT_SIZE; i++) {
+
+        input_tensor[i] = i;
+    }
+
+    float *output =
+        my_inference(input_tensor, dim, &output_size, "models/average.tflite");
+
+    printf("average is: %f \n ", output[0]);
+
+    assert(abs(output[0] - 12.0) < EPSILON);
+}
+
+void
+test_mult_dimensions()
+{
+    int output_size = 0;
+    uint32_t *dim = malloc(4 * sizeof(uint32_t));
+
+    dim[0] = 1;
+    dim[1] = 3;
+    dim[2] = 3;
+    dim[3] = 1;
+
+    float *input_tensor = malloc(9 * sizeof(float));
+
+    for (int i = 0; i < 9; i++) {
+
+        input_tensor[i] = i;
+    }
+
+    float *output =
+        my_inference(input_tensor, dim, &output_size, "models/mult_dim.tflite");
+
+    for (int i = 0; i < 9; i++) {
+        printf(" %f  \n ", output[i]);
+    }
```

Comment:
missing assert and put the print within a `#ifdef`

Reply:
done to be pushed